src/js/components/visualization/Visualization.jsx

```diff
@@ -93,6 +93,68 @@ class Visualization extends Component {
       typeGroup
     }
   }
+  
+  renderError (error) {
+    const {visualization} = this.props
+    const {name} = visualization
+    let text = error.message
+    
+    try {
+      text = JSON.parse(error.message)
+    } catch (err) {}
+    
+    return (
+      <div style={style.container}>
+        <VisualizationToolbar
+          title={name}
+          onClose={this.onClose}
+        />
+        <div style={{flex: 1, position: 'relative'}}>
+          <span
+            style={{
+              color: 'red',
+              left: '50%',
+              position: 'absolute',
+              top: '50%',
+              transform: 'translate(-50%, -50%)'
+            }}
+          >
+            {(() => {
+              if (typeof text === 'string') {
+                return (
+                  <div>
+                    {text}
+                  </div>
+                )
+              }
+                  
+              if (typeof text === 'object') {
+                return (
+                  <div>
+                    {text.label}
+                    {
+                      text.data.map((item) => {
```

Comment:
what does this data structure look like? why is data an array, but then we also pick off all the keys and iterate through them? couldn't data just be an object than?